js/stdlib/ts/ALGO.ts

```diff
@@ -1199,6 +1199,34 @@ const getApplicationInfoM = async (idn:BigNumber): Promise<OrExn<AppInfo>> => {
   return 'exn' in res ? res : { val: res.val.application };
 };
 
+const getDeletedApplicationInfoM = async (idn: BigNumber): Promise<OrExn<AppInfo>> => {
+  const dhead = 'getDeletedApplicationInfoM'
+  const indexer = await getIndexer();
+  const query = indexer.searchForTransactions()
+                       .txType('appl')
+                       .applicationID(bigNumberToNumber(idn))
+                       .limit(1);
+  const queryRes = await doQueryM_(dhead, query);
+
+  if ('val' in queryRes) {
```

Comment:
This isn't precise enough... you need to ensure that the transaction you found is actually the creation one. You can tell because the application id will be 0 and the created application id will be the one you searched for